[core]  One snapshot should have only one auto tag (#5832)
apache/paimon: docs/content/spark/sql-ddl.md

-- create or replace a tag, create tag if it not exist, replace tag if it exists.
ALTER TABLE T CREATE OR REPLACE TAG `TAG-5` AS OF VERSION 2 RETAIN 24 HOURS;
```

> > NOTE: If the table's property tag.automatic-creation is not none, which is default, only one auto-tag could be created for one snapshot. 

Reviewer: ```suggestion
NOTE: If tag auto-creation is set, only one auto-tag could be created for one snapshot. 
```
